polkadot/node/network/collator-protocol/src/validator_side_experimental/mod.rs

```diff
@@ -68,25 +68,30 @@ async fn initialize<Context>(
 ) -> FatalResult<Option<State<Db>>> {
 	loop {
 		let first_leaf = match wait_for_first_leaf(ctx).await? {
-			Some(activated_leaf) => activated_leaf,
+			Some(activated_leaf) => {
+				gum::debug!(
+					target: LOG_TARGET,
+					number = activated_leaf.number,
+					hash = ?activated_leaf.hash,
+					"Got the first active leaf notification, trying to initialize subsystem."
+				);
+				activated_leaf
+			},
 			None => return Ok(None),
 		};
 
-		let scheduled_paras = match scheduled_paras(ctx.sender(), first_leaf.hash, &keystore).await
-		{
-			Ok(paras) => paras,
-			Err(err) => {
-				log_error(Err(err))?;
-				continue
-			},
-		};
+		let collation_manager =
+			CollationManager::new(ctx.sender(), keystore.clone(), first_leaf).await?;
+
+		let scheduled_paras = collation_manager.assignments();
```

Comment:
Can this be empty on startup ? In ConnectedPeers::new() we use the count of scheduled para as divisor IIUC.

Reply:
Yes, but in `ConnectedPeers::new()` we were using `checked_div` which should be safe. Also reorganized the logic in `ConnectedPeers::new()` a bit.